### DISCUSSION TITLE
Using alternative collision filtering with CharacterVirtual?

### DISCUSSION THREAD

**Discussion by dxu** [jrouwe/JoltPhysics.js#180] [General]
Is it possible to use the alternative collision filtering with CharacterVirtual? I followed the example, but the character doesn't seem to be affected by any collisions. No ContactListener is triggered, and the simulation doesn't seem to register them colliding at all. All non-character bodies seem to collide with each other just fine

My code is largely derived from a combination of the examples, though I may be doing something wrong. I put what I thought might be the relevant code snippets in a gist: https://gist.github.com/dxu/f57e35d65a7349a6261904ec144b544d. Let me know if I need to include other code samples!

**Comment by jrouwe** [comment]
The alternative collision filtering should just work with a `CharacterVirtual`.

Note that `CharacterVirtual` doesn't trigger any callbacks on the `ContactListener` (it has no rigid body), instead it has callbacks on the `CharacterContactListenerJS` which you need to register through `CharacterVirtual.SetListener`.

I don't immediately see what's wrong with the code snippet (too many things are missing, I would need something that I can run). Perhaps you can build a Debug build of the library to see if there are any warnings/errors that trigger?

**Reply by dxu** [reply]
Thank you for the response! I wasn't able to build a Debug build on my system (running into cmake/emscripten issues on Mac, still figuring that out), but I was able to manually copy over builds and modify the example locally. 

It turns out I was simply not specifying the correct layers in my objectLayerFilter, I needed to generate the proper layer via `JOLT.ObjectLayerPairFilterMask.prototype.sGetObjectLayer`


```
objectLayer = JOLT.ObjectLayerPairFilterMask.prototype.sGetObjectLayer(..., ...)
  
new JOLT.DefaultObjectLayerFilter(
      joltInterface.GetObjectLayerPairFilter(),
      // objectFilter,
      objectLayer,
    );
```

Sorry for the churn!
